ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/LoadingPaywallView.kt

```diff
@@ -0,0 +1,148 @@
+        mode,
+        paywallData,
+        LoadingPaywallConstants.template,
+    )
+
+    when (state) {
+        is PaywallViewState.Error,
+        is PaywallViewState.Loading,
+        -> Box {}
+        is PaywallViewState.Loaded -> Template2(
+            state = state,
+            viewModel = viewModel,
+        )
+    }
+}
+
+private object LoadingPaywallConstants {
+    const val offeringIdentifier = "loading_offering"
+
+    val template = PaywallTemplate.TEMPLATE_2
+
+    val packages = listOf(
+        Package(
+            identifier = "weekly",
+            packageType = PackageType.WEEKLY,
+            offering = this.offeringIdentifier,
+            product = TestStoreProduct(
+                id = "com.revenuecat.weekly",
+                title = "Weekly",
+                price = Price(formatted = "$1.99", currencyCode = "USD", amountMicros = 1_990_000),
+                description = "Weekly",
+                period = Period(value = 1, unit = Period.Unit.WEEK, iso8601 = "P1W"),
+            ),
+        ),
+        Package(
+            identifier = "monthly",
+            packageType = PackageType.MONTHLY,
+            offering = this.offeringIdentifier,
+            product = TestStoreProduct(
+                id = "com.revenuecat.monthly",
+                title = "Monthly",
+                price = Price(formatted = "$5.99", currencyCode = "USD", amountMicros = 5_990_000),
+                description = "Monthly",
+                period = Period(value = 1, unit = Period.Unit.MONTH, iso8601 = "P1M"),
+            ),
+        ),
+        Package(
+            identifier = "annual",
+            packageType = PackageType.ANNUAL,
+            offering = this.offeringIdentifier,
+            product = TestStoreProduct(
+                id = "com.revenuecat.annual",
+                title = "Annual",
+                price = Price(formatted = "$15.99", currencyCode = "USD", amountMicros = 5_990_000),
+                description = "Annual",
+                period = Period(value = 1, unit = Period.Unit.YEAR, iso8601 = "P1Y"),
+            ),
+        ),
+    )
+}
+
+private class LoadingViewModel(
+    mode: PaywallViewMode = PaywallViewMode.default,
+    offering: Offering,
+    paywall: PaywallData,
+) : ViewModel(), PaywallViewModel {
+    override val state: StateFlow<PaywallViewState>
+        get() = _state.asStateFlow()
+
+    override fun refreshStateIfLocaleChanged() = Unit
+    override fun refreshStateIfColorsChanged(colorScheme: ColorScheme) = Unit
+
+    private val _state =
+        MutableStateFlow(
+            offering.toPaywallViewState(
+                VariableDataProvider(MockApplicationContext()),
+                mode,
+                template = LoadingPaywallConstants.template,
+                validatedPaywallData = paywall,
+            ),
+        )
+
+    override fun selectPackage(packageToSelect: TemplateConfiguration.PackageInfo) {
+        error("Not supported")
```

Comment:
Hmm what would happen if people tapped on these buttons while they are loading? Would it crash? I think we probably want to do nothing on these.